examples/recognition_pipeline/recognition_pipeline.py

```diff
@@ -0,0 +1,198 @@
+
+    """
+    def setup(self):
+        self.db = initialize_DB()
+
+    def process(self, element, run_id):
+        """Updates the project's database to contain documents with the currect fields for each label in the Labels subcollection of each image.
+
+        Args: 
+            element: tuple of image document dict and a list of all label Ids.
+        """
+        # TODO: need to figure out how to add the pipeline run's id inside the pipeline
+        doc_id = element[0]['id']
+        subcollection_ref = self.db.collection(collection_name).document(doc_id).collection(u'Labels')
+        for label in element[1]:
+            doc = subcollection_ref.document()
+            doc.set({
+                u'providerId': 'google_vision_api',
+                u'providerVersion': '2.0.0',
+                u'labelName': label['name'],
+                u'visibility': 0,
+                u'parentImageId': doc_id,
+                u'pipelineRunId': run_id,
+                u'hashmap': element[0]['hashmap']
+            })
+            if 'id' in label:
+                doc.set({
+                    u'labelId': label['id']
+                }, merge = True)
+
+def upload_to_pipeline_runs_collection(provider_Id, run_Id):
+    """ Uploads inforamtion about the pipeline run to the Firestore collection
+    """
+    # TODO: get start, end and quality of current pipeline run.
+    db = initialize_DB()
+    db.collection(u'RecognitionPipelineRuns').document().set({
+        u'providerId': provider_Id,
+        u'startDate': 00,
+        u'endDate': 00,
+        u'quality': 00,
+        u'visibility': 0,
+        u'pipelineRunId': run_Id
+    })
+
+def run(argv=None, save_main_session=True):
+  """Main entry point, defines and runs the image recognition pipeline."""
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--input-ingestion-run',
+      dest='input_ingestion_run',
+      help='Input of ingestion pipeline run for images dataset.')
+  parser.add_argument(
+      '--input-ingestion-provider',
+      dest='input_ingestion_provider',
+      help='Input of ingestion pipeline provider for images dataset.')
+  parser.add_argument(
+      '--input-recognition-provider',
+      dest='input_recognition_provider',
+      required=True,
+      default='Google_Vision_API',
+      help='Input fimage recognition provider for labeling.')
+  parser.add_argument(
+      '--output',
+      dest='output',
+      help='Output file to write results to.')
+  known_args, pipeline_args = parser.parse_known_args(argv)
+  job_name = 'recognition-pipeline-run-{date_time}'.format(date_time = datetime.now())
+ 
+  pipeline_options = PipelineOptions(pipeline_args)
+
+  with beam.Pipeline(options=pipeline_options) as p:
+    ingestion_run = known_args.input_ingestion_run
+    ingestion_provider = known_args.input_ingestion_provider
+    provider = get_provider(known_args.input_recognition_provider)
```

Comment:
rename to `label_provider` for clarity

Reply:
renamed to recognition_provider to be consistent.